### PR TITLE
Fix the backup problem

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref DEV-1362

### CODE REVIEW

File: android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt [hunk 1]
```diff
@@ -367,17 +373,63 @@ class AuthgearPlugin: FlutterPlugin, ActivityAware, MethodCallHandler, PluginReg
...
-      )
+      try {
+        return EncryptedSharedPreferences.create(
+          context,
+          ENCRYPTED_SHARED_PREFERENCES_NAME,
+          masterKey,
+          EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+          EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        )
+      } catch (e: InvalidProtocolBufferException) {
```
Comment (tung2744) [inline android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt:427]: I suggest we catch both `IOException` and `GeneralSecurityException` here, which are documented to be possible exceptions of `EncryptedSharedPreferences.create`.
https://developer.android.com/reference/androidx/security/crypto/EncryptedSharedPreferences#create(android.content.Context,java.lang.String,androidx.security.crypto.MasterKey,androidx.security.crypto.EncryptedSharedPreferences.PrefKeyEncryptionScheme,androidx.security.crypto.EncryptedSharedPreferences.PrefValueEncryptionScheme)

And it seems calling `deleteSharedPreferences` is always make sense when this method throwed execption.

File: android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt [hunk 2]
```diff
@@ -388,6 +440,15 @@ class AuthgearPlugin: FlutterPlugin, ActivityAware, MethodCallHandler, PluginReg
       sharedPreferences.edit().putString(key, value).commit()
       result.success(null)
     } catch (e: Exception) {
+      // NOTE(backup): Please search NOTE(backup) to understand what is going on here.
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+        if (e is AEADBadTagException) {
```
Comment (tung2744) [inline android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt:445]: Suggest to catch `GeneralSecurityException` here too.

File: android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt (hunk 1 above)
Comment (louischan-oursky) [reply android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt:427]: Fixed.

File: android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt (hunk 2 above)
Comment (louischan-oursky) [reply android/src/main/kotlin/com/authgear/flutter/AuthgearPlugin.kt]: Fixed.

**Comment by louischan-oursky** [comment]
One thing is missing in this PR. The example app should support the newer backup mechanism.

**Comment by louischan-oursky** [comment]
The example app now supports both Android <= 11 and Android >= 12 backup mechanism.

**Pull request closed without merging**